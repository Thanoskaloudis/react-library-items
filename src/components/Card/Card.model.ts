export interface ICard {
  title: string
  imgURL: string
  duration: string
  type: string
  link: string
  description: string
}