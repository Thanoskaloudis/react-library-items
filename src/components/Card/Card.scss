.card {
  background: white;
  display: inline-block;
  margin: 1rem;
  max-width: 20rem;
  text-align: left;

  img {
    max-width: 20rem;
    border-radius: 0.8rem;  
  }
}