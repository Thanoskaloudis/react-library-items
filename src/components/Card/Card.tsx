import React from 'react';
import { ICard } from './Card.model';

export const Card = (props: ICard) => {
  return (
    <div className="card">
      <div className="card__image-holder">
        <img
          className="card__image"
          src={props.imgURL}
          alt=""
        />
      </div>
      <div className="card-title">
        <h2>{props.title}</h2>
      </div>
    </div>
  );
};
