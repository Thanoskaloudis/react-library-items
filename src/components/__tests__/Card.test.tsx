import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Card } from '../Card/Card';

describe('<Selection />', () => {
  let mockTitle = 'test';
  let mockImgURL = 'image test url'
  let mockDuration = '10 mins'
  let mockType = 'test pathway';
  let mockLink = 'test link';
  let mockDescription = 'test description';

  const component = renderer.create(<Card
            title={mockTitle}
            imgURL={mockImgURL}
            duration={mockDuration}
            type={mockType}
            link={mockLink}
            description={mockDescription}
        />);

  it('Should render correctly', () => {
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should have the correct image url', () => {
    render(<Card
      title={mockTitle}
      imgURL={mockImgURL}
      duration={mockDuration}
      type={mockType}
      link={mockLink}
      description={mockDescription}
  />);

    const image: HTMLImageElement = screen.getByRole('img', {});

    expect(image).toHaveAttribute('src', mockImgURL);
  });

  it('should render the header and required input', () => {
    render(<Card
      title={mockTitle}
      imgURL={mockImgURL}
      duration={mockDuration}
      type={mockType}
      link={mockLink}
      description={mockDescription}
  />);

    expect(screen.getByTestId('cardTitle')).toBeTruthy();
    expect(screen.getByText(mockTitle)).toBeTruthy();
  });

  it('should render card with the correct duration and type', () => {
    render(<Card
      title={mockTitle}
      imgURL={mockImgURL}
      duration={mockDuration}
      type={mockType}
      link={mockLink}
      description={mockDescription}
  />);

  const subTitle = screen.getByTestId('cardSubTitle');

  expect(subTitle.textContent).toEqual(`  ${mockType} â€¢ ${mockDuration}`);
  });

  it('should render card with the correct card link', () => {
    render(<Card
      title={mockTitle}
      imgURL={mockImgURL}
      duration={mockDuration}
      type={mockType}
      link={mockLink}
      description={mockDescription}
  />);

  const link = screen.getByTestId('cardLink');

  expect(link).toHaveAttribute('href', mockLink);
  });
});
