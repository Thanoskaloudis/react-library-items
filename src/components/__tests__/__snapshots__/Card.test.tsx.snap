// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Selection /> Should render correctly 1`] = `
<div
  className="card"
>
  <div
    className="card--image"
  >
    <img
      alt=""
      src="image test url"
    />
  </div>
  <div
    className="card--title"
  >
    <span
      className="card--title__subtitle"
      data-testid="cardSubTitle"
    >
       
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M21 18H6a1 1 0 0 0 0 2h15v2H6a3 3 0 0 1-3-3V4a2 2 0 0 1 2-2h16v16zm-5-9V7H8v2h8z"
          />
        </g>
      </svg>
       
      test pathway
       â€¢ 
      10 mins
    </span>
    <h2
      className="card--title__title"
      data-testid="cardTitle"
    >
      test
    </h2>
  </div>
  <div
    className="card--description"
  >
    test description
  </div>
  <div
    className="card--link"
  >
    <a
      className="card--link__href"
      data-testid="cardLink"
      href="test link"
      rel="noopener noreferrer"
      target="_blank"
    >
      View pathway
    </a>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
      />
    </svg>
  </div>
</div>
`;
